### PR TITLE
CSVRecordReader – Set header record when header is explicitly provided

### PR DESCRIPTION
**Pull request by rajagopr**
Fixing regression caused by the previous changes:
1. Set header record when the header record is supplied by the caller in the default CSVRecordReader
2. When header record is not provided always treat the first record as the header record

Added new tests to cover the changes.

### CODE REVIEW

**Review by snleee — APPROVED** [review]
LGTM

File: pinot-plugins/pinot-input-format/pinot-csv/src/test/resources/dataFileWithNoHeader2.csv [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"key00",12.3,8.42,17.86,-11.52,,"FURNITURE","CLAMP ON LAMPS","one","e",1,-3,5,8,1,1,0,1,"2004-04-15 00:00:00","2004-04-01 00:00:00","1977-04-20 00:00:00","1986-03-20 00:00:00","1899-12-30 21:07:32","1900-01-01 19:36:22","2004-07-09 10:17:35",,"a"
```
Comment (shenyu0127) [inline pinot-plugins/pinot-input-format/pinot-csv/src/test/resources/dataFileWithNoHeader2.csv:1]: We can use a simpler table with one column to make the test easier to read.

**Comment by shenyu0127** [comment]
Thank you @rajagopr for the quick fix!

I left one comment and everything else looks good to me.

**Review by shenyu0127 — APPROVED** [review]
LGTM!

**Pull request merged**